desc/objectives/_reconstruction.py

```diff
@@ -1,18 +1,56 @@
 """Objectives for experimental reconstruction problems."""
 
+from abc import ABC, abstractmethod
+
 import numpy as np
 
 from desc.backend import jnp
 from desc.compute import get_profiles, get_transforms, xyz2rpz, xyz2rpz_vec
 from desc.compute.utils import _compute as compute_fun
 from desc.grid import LinearGrid
-from desc.utils import Timer, dot, errorif
+from desc.utils import Timer, dot, errorif, warnif
 
 from .normalization import compute_scaling_factors
 from .objective_funs import _Objective, collect_docs
 
 
-class PointBMeasurement(_Objective):
+class _MagneticMeasurement(_Objective, ABC):
```

Comment:
For new stuff I've been pushing to name things like `AbstractMagneticMeasurement` rather than an underscore to make clear that its an ABC, and that it maybe semi-public (if users want to add their own subclasses)